planner/core/find_best_task.go

```diff
@@ -205,11 +168,76 @@ func (p *baseLogicalPlan) findBestTask(prop *property.PhysicalProperty) (bestTas
 		}
 
 		// get the most efficient one.
-		if curTask.cost() < bestTask.cost() {
+		if curTask.cost() < bestTask.cost() || (bestTask.invalid() && !curTask.invalid()) {
 			bestTask = curTask
 		}
 	}
+	return bestTask, nil
+}
+
+// findBestTask implements LogicalPlan interface.
+func (p *baseLogicalPlan) findBestTask(prop *property.PhysicalProperty) (bestTask task, err error) {
```

Comment:
A unit test for this method is appreciated.

Reply:
I'm working on this, it need some time...